### PR TITLE
chore: adds cargo deny workflow

### PR DESCRIPTION
**Pull request by montekki**
I think it would be a good idea to check at least the advisories.

### CODE REVIEW

**Comment by Slesarew** [comment]
Let's fix and merge this after #962, merging anything there becomes too challenging already

File: .github/workflows/rust-cargo-deny.yml
Comment (sergejparity) [inline .github/workflows/rust-cargo-deny.yml:23]:
```suggestion
        uses:               actions/checkout@v3
```
I suggest to use latest version of this GHA

**Review by Slesarew — APPROVED** [review]

**Pull request merged**
